pkg/server/tenant_admin.go

```diff
@@ -84,4 +99,52 @@ func (t *tenantAdminServer) Health(
 	return resp, nil
 }
 
-// TODO(knz): add Drain implementation here.
+func (t *tenantAdminServer) dialPod(
+	ctx context.Context, instanceID base.SQLInstanceID, addr string,
+) (serverpb.AdminClient, error) {
+	conn, err := t.sqlServer.execCfg.RPCContext.GRPCDialPod(addr, instanceID, rpc.DefaultClass).Connect(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// nb: The server on the pods doesn't implement all the methods of the
+	// `AdminService`. It is up to the caller of `dialPod` to only call
+	// methods that are implemented on the tenant server.
+	return serverpb.NewAdminClient(conn), nil
+}
+
+// Drain puts the node into the specified drain mode(s) and optionally
+// instructs the process to terminate.
+// This method is part of the serverpb.AdminClient interface.
+func (t *tenantAdminServer) Drain(
+	req *serverpb.DrainRequest, stream serverpb.Admin_DrainServer,
+) error {
+	ctx := stream.Context()
+	ctx = t.AnnotateCtx(ctx)
+
+	// Which node is this request for?
+	parsedInstanceID, local, err := t.status.parseInstanceID(req.NodeId)
```

Comment:
FYI: We do not track instance ID in the CC operator. The operator thinks in terms of pods. Determining the instance ID of the pod we want to remove is more difficult than sending an RPC to the specific pod we want to drain. I suspect we will pass NodeId == "" when sending a Drain rpc.